pkg/ccl/sqlccl/load.go

```diff
@@ -0,0 +1,368 @@
+			tableDescs[tableName] = tableDesc
+			backup.Descriptors = append(backup.Descriptors, sqlbase.Descriptor{
+				Union: &sqlbase.Descriptor_Table{Table: tableDesc},
+			})
+
+			ri, err = sql.MakeRowInserter(nil, tableDesc, nil, tableDesc.Columns, true)
+			if err != nil {
+				return BackupDescriptor{}, errors.Wrap(err, "make row inserter")
+			}
+			cols, defaultExprs, err = sql.ProcessDefaultColumns(tableDesc.Columns, tableDesc, &parse, &evalCtx)
+			if err != nil {
+				return BackupDescriptor{}, errors.Wrap(err, "process default columns")
+			}
+
+		case *parser.Insert:
+			name := parser.AsString(s.Table)
+			if tableDesc == nil {
+				return BackupDescriptor{}, errors.Errorf("expected previous CREATE TABLE %s statement", name)
+			}
+			if parser.ReNormalizeName(name) != parser.ReNormalizeName(tableName) {
+				return BackupDescriptor{}, errors.Errorf("unexpected INSERT for table %s after CREATE TABLE %s", name, tableName)
+			}
+			outOfOrder := false
+			err := insertStmtToKVs(ctx, tableDesc, defaultExprs, cols, evalCtx, ri, s, func(kv roachpb.KeyValue) {
+				if outOfOrder || prevKey.Compare(kv.Key) >= 0 {
+					outOfOrder = true
+					return
+				}
+				prevKey = kv.Key
+				kvBytes += int64(len(kv.Key) + len(kv.Value.RawBytes))
+				kvs = append(kvs, engine.MVCCKeyValue{
+					Key:   engine.MVCCKey{Key: kv.Key, Timestamp: kv.Value.Timestamp},
+					Value: kv.Value.RawBytes,
+				})
+			})
+			if err != nil {
+				return BackupDescriptor{}, errors.Wrapf(err, "insertStmtToKVs")
+			}
+			if outOfOrder {
+				return BackupDescriptor{}, errors.Errorf("out of order row: %s", cmd)
+			}
+
+			if kvBytes > loadChunkBytes {
+				if err := writeSST(ctx, &backup, dir, kvs, ts); err != nil {
+					return BackupDescriptor{}, errors.Wrap(err, "writeSST")
+				}
+				kvs = kvs[:0]
+				kvBytes = 0
+			}
+
+		default:
+			return BackupDescriptor{}, errors.Errorf("unsupported load statement: %q", stmt)
+		}
+	}
+
+	if tableDesc != nil {
+		if err := writeSST(ctx, &backup, dir, kvs, ts); err != nil {
+			return BackupDescriptor{}, errors.Wrap(err, "writeSST")
+		}
+	}
+
+	descBuf, err := backup.Marshal()
+	if err != nil {
+		return BackupDescriptor{}, errors.Wrap(err, "marshal backup descriptor")
+	}
+	descFile, err := dir.PutFile(ctx, BackupDescriptorName)
+	if err != nil {
+		return BackupDescriptor{}, errors.Wrap(err, "creating backup descriptor file")
+	}
+	defer descFile.Cleanup()
+	if err = ioutil.WriteFile(descFile.LocalFile(), descBuf, 0600); err != nil {
+		return BackupDescriptor{}, errors.Wrap(err, "write backup descriptor")
+	}
+	if err := descFile.Finish(); err != nil {
+		return BackupDescriptor{}, errors.Wrap(err, "uploading backup descriptor")
+	}
+
+	return backup, nil
+}
+
+func insertStmtToKVs(
+	ctx context.Context,
+	tableDesc *sqlbase.TableDescriptor,
+	defaultExprs []parser.TypedExpr,
+	cols []sqlbase.ColumnDescriptor,
+	evalCtx parser.EvalContext,
+	ri sql.RowInserter,
+	stmt *parser.Insert,
+	f func(roachpb.KeyValue),
```

Comment:
Could use some documentation as to what `f` is. Actually, it would make things easier if we defined this inside `insertStmtToKVs` and instead pass and return `kvs`.